Add support for text indicator style.

megangardner opened the pull request:
#### cfd62a6c05a99c28824420b9cd9bd1d251bb868d
<pre>
Add support for text indicator style.
<a href="https://bugs.webkit.org/show_bug.cgi?id=271877">https://bugs.webkit.org/show_bug.cgi?id=271877</a>
<a href="https://rdar.apple.com/123134420">rdar://123134420</a>

Reviewed by Aditya Keerthi.

Add support for text indicator style.

* Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
* Source/WebKit/Configurations/WebKitSwift.xcconfig:
* Source/WebKit/DerivedSources-input.xcfilelist:
* Source/WebKit/DerivedSources-output.xcfilelist:
* Source/WebKit/DerivedSources.make:
* Source/WebKit/UIProcess/API/Cocoa/WKWebView.mm:
(-[WKWebView _removeStyleForID:]):
* Source/WebKit/UIProcess/API/Cocoa/WKWebViewInternal.h:
* Source/WebKit/UIProcess/Cocoa/PageClientImplCocoa.h:
* Source/WebKit/UIProcess/Cocoa/PageClientImplCocoa.mm:
(WebKit::PageClientImplCocoa::removeStyleForID):
* Source/WebKit/UIProcess/Cocoa/WebKitSwiftSoftLink.h:
* Source/WebKit/UIProcess/Cocoa/WebKitSwiftSoftLink.mm:
* Source/WebKit/UIProcess/Cocoa/WebPageProxyCocoa.mm:
(WebKit::WebPageProxy::getTextIndicatorForID):
(WebKit::WebPageProxy::updateTextStyleVisibilityForID):
* Source/WebKit/UIProcess/PageClient.h:
* Source/WebKit/UIProcess/WebPageProxy.cpp:
(WebKit::WebPageProxy::removeStyleForID):
* Source/WebKit/UIProcess/WebPageProxy.h:
* Source/WebKit/UIProcess/WebPageProxy.messages.in:
* Source/WebKit/UIProcess/ios/WKContentView.h:
* Source/WebKit/UIProcess/ios/WKContentViewInteraction.h:
* Source/WebKit/UIProcess/ios/WKContentViewInteraction.mm:
(-[WKContentView addStyleForID:]):
(-[WKContentView removeStyleForID:]):
(-[WKContentView targetedPreviewForID:completionHandler:]):
(-[WKContentView updateTextStyleVisibilityForID:visible:completionHandler:]):
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
* Source/WebKit/WebKitSwift/TextIndicatorStyle/WKSTextStyleManager.h: Copied from Source/WebKit/WebKitSwift/WebKitSwift.h.
* Source/WebKit/WebKitSwift/WebKitSwift.h:
* Source/WebKit/WebProcess/WebCoreSupport/WebChromeClient.h:
* Source/WebKit/WebProcess/WebPage/Cocoa/UnifiedTextReplacementController.mm:
(WebKit::UnifiedTextReplacementController::textReplacementSessionDidReceiveReplacements):
(WebKit::UnifiedTextReplacementController::textReplacementSessionDidReceiveTextWithReplacementRange):
* Source/WebKit/WebProcess/WebPage/Cocoa/WebPageCocoa.mm:
(WebKit::WebPage::getTextIndicatorForID):
(WebKit::WebPage::updateTextStyleVisibilityForID):
* Source/WebKit/WebProcess/WebPage/UnifiedTextReplacementController.h:
* Source/WebKit/WebProcess/WebPage/WebPage.cpp:
(WebKit::WebPage::removeStyleForID):
* Source/WebKit/WebProcess/WebPage/WebPage.h:
* Source/WebKit/WebProcess/WebPage/WebPage.messages.in:

Canonical link: <a href="https://commits.webkit.org/276943@main">https://commits.webkit.org/276943@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/bfe5821374b1ce744e362b38580728911923433a

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/46192 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/25323 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/48777 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/48864 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/42233 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/29680 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/22769 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/37743 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/46770 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/22414 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/39834 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/18952 "Passed tests") | 
| | [❌ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/19803 "Found unexpected failure with change (failure)") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/40932 "Passed tests") | [✅ 🛠 wpe-skia](https://ews-build.webkit.org/#/builders/52/builds/4236 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/39425 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/42511 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/41281 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/50666 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/45666 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/21188 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/17677 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/44927 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/22488 "Built successfully") | | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/43842 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/10235 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/22847 "Built successfully") | | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/52817 "Built successfully") | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/22182 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/10813 "Passed tests") | 
<!--EWS-Status-Bubble-End-->

rr-codes commented on Source/WebKit/UIProcess/Cocoa/PageClientImplCocoa.h:
```diff
@@ -96,6 +96,10 @@ class PageClientImplCocoa : public PageClient {
 #endif
 
+#if ENABLE(UNIFIED_TEXT_REPLACEMENT)
+    void removeStyleForID(const WTF::UUID&&) final;
```
r-values shouldn’t need `const` I think

whsieh replied:
(Indeed)

rr-codes commented on Source/WebKit/UIProcess/ios/WKContentView.h:
```diff
@@ -52,7 +52,6 @@ class WebProcessProxy;
 class WebProcessPool;
 enum class ViewStabilityFlag : uint8_t;
 }
-
```
should probably put this back

rr-codes commented on Source/WebKit/UIProcess/ios/WKContentViewInteraction.mm:
```diff
@@ -13061,6 +13090,42 @@ - (BOOL)shouldSuppressEditMenu
+    _page->getTextIndicatorForID(*textUUID, [protectedSelf = retainPtr(self), completionHandler = makeBlockPtr(completionHandler)] (std::optional<WebCore::TextIndicatorData> indicatorData) {
+
+        if (!indicatorData)
+            completionHandler(nil);
```
can `nil` safety be converted to `std::nullopt`?

whsieh replied:
Where is nil being converted to null opt?

rr-codes commented on Source/WebKit/UIProcess/ios/WKContentViewInteraction.mm:
```diff
@@ -13061,6 +13090,42 @@ - (BOOL)shouldSuppressEditMenu
+    _page->getTextIndicatorForID(*textUUID, [protectedSelf = retainPtr(self), completionHandler = makeBlockPtr(completionHandler)] (std::optional<WebCore::TextIndicatorData> indicatorData) {
+
+        if (!indicatorData)
+            completionHandler(nil);
```
there should be a return after each completion handler call

rr-codes commented on Source/WebKit/WebProcess/WebPage/Cocoa/WebPageCocoa.mm:
```diff
@@ -301,6 +302,38 @@
+#if ENABLE(UNIFIED_TEXT_REPLACEMENT)
+void WebPage::getTextIndicatorForID(const WTF::UUID uuid, CompletionHandler<void(std::optional<WebCore::TextIndicatorData>&&)>&& completionHander)
+{
+    RefPtr<WebCore::Range> range = m_unifiedTextReplacementController->rangeForUUID(uuid);
```
just `RefPtr` is enough

rr-codes replied:
`rangeForUUID` could probably use a more descriptive name

rr-codes commented on Source/WebKit/WebProcess/WebPage/Cocoa/WebPageCocoa.mm:
```diff
@@ -301,6 +302,38 @@
+    if (!simpleRange)
+        completionHander(std::nullopt);
+
+    if (RefPtr localMainFrame = dynamicDowncast<LocalFrame>(m_page->mainFrame())) {
```
`m_page` should be checked beforehand to make sure it’s not null

rr-codes commented on Source/WebKit/WebProcess/WebPage/Cocoa/WebPageCocoa.mm:
```diff
@@ -301,6 +302,38 @@
+    auto simpleRange = makeSimpleRange(range.get());
+
+    if (!simpleRange)
+        completionHander(std::nullopt);
```
missing early return

rr-codes commented on Source/WebKit/WebProcess/WebPage/WebPage.messages.in:
```diff
@@ -375,6 +375,11 @@ GenerateSyntheticEditingCommand(enum:uint8_t WebKit::SyntheticEditingCommandType
     DidConcludeDrop()
 #endif
 
+# ENABLE(UNIFIED_TEXT_REPLACEMENT)
```
is this missing an `if`

rr-codes commented on Source/WebKit/WebProcess/WebPage/Cocoa/UnifiedTextReplacementController.mm:
```diff
@@ -416,6 +418,8 @@ static void replaceContentsInRange(WebCore::Document& document, const WebCore::S
         return;
     }
 
+    m_webPage->removeStyleForID(WTFMove(uuid));
```
this uuid shouldn’t be moved; I use it later on in this method

rr-codes commented on Source/WebKit/WebProcess/WebPage/Cocoa/UnifiedTextReplacementController.mm:
```diff
@@ -235,6 +235,8 @@ static void replaceContentsInRange(WebCore::Document& document, const WebCore::S
         return;
     }
 
+    m_webPage->removeStyleForID(WTFMove(uuid));
```
ditto

pxlcoder commented on Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
```diff
@@ -6476,6 +6476,21 @@ TextExtractionEnabled:
+TextIndicatorStylingEnabled:
+  type: bool
+  status: internal
+  category: css
```
Actually, we can just omit `category` entirely.

pxlcoder commented on Source/WebKit/UIProcess/API/Cocoa/WKWebView.mm:
```diff
@@ -1765,6 +1765,15 @@ - (void)_storeAppHighlight:(const WebCore::AppHighlight&)highlight
 #endif
 
+#if ENABLE(UNIFIED_TEXT_REPLACEMENT)
+- (void)_removeStyleForID:(NSUUID *)uuid
```
`_removeTextIndicatorStyleForID`? `Style` could mean anything. Same comment for similar method names.

pxlcoder commented on Source/WebKit/UIProcess/Cocoa/WebPageProxyCocoa.mm:
```diff
@@ -288,6 +288,27 @@ static bool exceedsRenderTreeSizeSizeThreshold(uint64_t thresholdSize, uint64_t
+    sendWithAsyncReply(Messages::WebPage::GetTextIndicatorForID(uuid), WTFMove(completionHandler));
+}
+
+void WebPageProxy::updateTextStyleVisibilityForID(WTF::UUID& uuid, bool visible, CompletionHandler<void()>&& completionHandler)
```
It seems that `Style`, `TextIndicatorStyle`, and `TextStyle` are used inconsistently.

pxlcoder commented on Source/WebKit/UIProcess/ios/WKContentViewInteraction.mm:
```diff
@@ -13061,6 +13090,54 @@ - (BOOL)shouldSuppressEditMenu
+            return;
+        }
+
+        auto image = adoptNS([[UIImage alloc] initWithCGImage:snapshotImage->platformImage().get() scale:protectedSelf->_page->deviceScaleFactor() orientation:UIImageOrientationUp]);
```
`auto` -> `RetainPtr` due to explicit smart point usage guidelines.

pxlcoder commented on Source/WebKit/UIProcess/ios/WKContentViewInteraction.mm:
```diff
@@ -13061,6 +13090,54 @@ - (BOOL)shouldSuppressEditMenu
+
+        auto image = adoptNS([[UIImage alloc] initWithCGImage:snapshotImage->platformImage().get() scale:protectedSelf->_page->deviceScaleFactor() orientation:UIImageOrientationUp]);
+
+        auto targetedPreview = createTargetedPreview(image.get(), protectedSelf.get(), [protectedSelf containerForContextMenuHintPreviews], indicatorData->textBoundingRectInRootViewCoordinates, indicatorData->textRectsInBoundingRectCoordinates, nil);
```
Ditto re: `auto`.

pxlcoder commented on Source/WebKit/WebProcess/WebCoreSupport/WebChromeClient.h:
```diff
@@ -387,7 +387,6 @@ class WebChromeClient final : public WebCore::ChromeClient {
 #if ENABLE(APP_HIGHLIGHTS)
     void storeAppHighlight(WebCore::AppHighlight&&) const final;
 #endif
-
```
This change is not relevant to this PR.

pxlcoder commented on Source/WebKit/WebProcess/WebPage/Cocoa/WebPageCocoa.mm:
```diff
@@ -301,6 +302,45 @@
 }
 
+#if ENABLE(UNIFIED_TEXT_REPLACEMENT)
+void WebPage::getTextIndicatorForID(const WTF::UUID uuid, CompletionHandler<void(std::optional<WebCore::TextIndicatorData>&&)>&& completionHander)
```
Type: `completionHander`

rr-codes replied:
should be `const WTF::UUID& uuid` (note the `&`)

rr-codes replied:
I think maybe all of this logic should be moved to the unified text replacement controller

pxlcoder commented on Source/WebKit/WebProcess/WebPage/UnifiedTextReplacementController.h:
```diff
@@ -66,6 +66,8 @@ class UnifiedTextReplacementController final {
 
     void textReplacementSessionDidReceiveEditAction(const WTF::UUID&, WebKit::WebTextReplacementData::EditAction);
 
+    RefPtr<WebCore::Range> rangaAssociatedWithUUID(const WTF::UUID uuid) { return m_contextRanges.get(uuid); }
```
Typo in the current method name. However, can we call this `contextRangeForUUID` instead?

rr-codes replied:
I think all the logic in webpage should be moved inside the unified text replacement controller, and then this method can be removed entirely

pxlcoder replied:
+1 to that.

whsieh commented on Source/WebKit/WebProcess/WebPage/WebPage.messages.in:
```diff
@@ -375,6 +375,11 @@ GenerateSyntheticEditingCommand(enum:uint8_t WebKit::SyntheticEditingCommandType
+#if ENABLE(UNIFIED_TEXT_REPLACEMENT)
+    GetTextIndicatorForID(WTF::UUID uuid) -> (std::optional<WebCore::TextIndicatorData> textIndicator)
+    UpdateTextStyleVisibilityForID(WTF::UUID uuid, bool visible) -> ()
+# endif
```
There's an extra space after the `#` here, which is causing the EWS builds to fail (since it can't find definitions for the subsequent IPC messages).

whsieh commented on Source/WebKit/WebProcess/WebPage/Cocoa/WebPageCocoa.mm:
```diff
@@ -301,6 +302,45 @@
+            textIndicatorData = textIndicator->data();
+
+        completionHander(WTFMove(textIndicatorData));
+    }
```
Make sure you call the `completionHandler` in the case where there's no `localMainFrame` as well.

whsieh commented on Source/WebKit/WebProcess/WebPage/Cocoa/WebPageCocoa.mm:
```diff
@@ -301,6 +302,45 @@
+
+void WebPage::updateTextStyleVisibilityForID(const WTF::UUID uuid, bool visible, CompletionHandler<void()>&& completionHandler)
+{
+    // TODO: turn on/off the visibility
```
Nit - `FIXME` is the convention in WebKit.

rr-codes commented on Source/WebKit/WebProcess/WebPage/WebPage.cpp:
```diff
@@ -9027,6 +9027,15 @@ void WebPage::lastNavigationWasAppInitiated(CompletionHandler<void(bool)>&& comp
     return completionHandler(mainFrame->document()->loader()->lastNavigationWasAppInitiated());
 }
 
+#if ENABLE(UNIFIED_TEXT_REPLACEMENT)
```
this guard isn’t needed

rr-codes replied:
nvm

rr-codes commented on Source/WebKit/WebProcess/WebPage/Cocoa/UnifiedTextReplacementController.mm:
```diff
@@ -699,6 +704,42 @@ static void replaceContentsInRange(WebCore::Document& document, const WebCore::S
+    }
+}
+
+
```
so many newlines!

rr-codes commented on Source/WebKit/WebProcess/WebPage/Cocoa/UnifiedTextReplacementController.mm:
```diff
@@ -699,6 +704,42 @@ static void replaceContentsInRange(WebCore::Document& document, const WebCore::S
+void UnifiedTextReplacementController::getTextIndicatorForID(const WTF::UUID& uuid, CompletionHandler<void(std::optional<WebCore::TextIndicatorData>&&)>&& completionHandler)
+{
+    RefPtr range = m_contextRanges.get(uuid);
+    auto simpleRange = makeSimpleRange(range.get());
```
this should be `WebCore::makeSimpleRange`

rr-codes commented on Source/WebKit/WebProcess/WebPage/Cocoa/UnifiedTextReplacementController.mm:
```diff
@@ -699,6 +704,42 @@ static void replaceContentsInRange(WebCore::Document& document, const WebCore::S
+        return;
+    }
+
+    if (RefPtr localMainFrame = dynamicDowncast<WebCore::LocalFrame>(corePage->mainFrame())) {
```
this should be inverted to just be an early return

rr-codes commented on Source/WebKit/WebProcess/WebPage/Cocoa/UnifiedTextReplacementController.mm:
```diff
@@ -699,6 +704,42 @@ static void replaceContentsInRange(WebCore::Document& document, const WebCore::S
+            WebCore::TextIndicatorOption::ExpandClipBeyondVisibleRect,
+            WebCore::TextIndicatorOption::UseSelectionRectForSizing
+        };
+        if (auto textIndicator = WebCore::TextIndicator::createWithRange(*simpleRange, textIndicatorOptions, WebCore::TextIndicatorPresentationTransition::None, { }))
```
ditto

pxlcoder commented on Source/WebKit/UIProcess/WebPageProxy.h:
```diff
@@ -1065,6 +1066,12 @@ class WebPageProxy final : public API::ObjectImpl<API::Object::Type::Page>, publ
+#if ENABLE(UNIFIED_TEXT_REPLACEMENT)
+    void getTextIndicatorForID(WTF::UUID&, CompletionHandler<void(std::optional<WebCore::TextIndicatorData>&&)>&&);
+    void updateTextIndicatorStyleVisibilityForID(WTF::UUID&, bool, CompletionHandler<void()>&&);
+
```
Nit: Stray newline.

pxlcoder commented on Source/WebKit/WebProcess/WebPage/Cocoa/UnifiedTextReplacementController.mm:
```diff
@@ -699,6 +704,48 @@ static void replaceContentsInRange(WebCore::Document& document, const WebCore::S
 }
 
+
+void UnifiedTextReplacementController::getTextIndicatorForID(const WTF::UUID& uuid, CompletionHandler<void(std::optional<WebCore::TextIndicatorData>&&)>&& completionHandler)
```
Is there a reason the completion handler takes an optional? Because of your early return on line 741, the optional will never be null.

pxlcoder replied:
Never mind 😅

pxlcoder commented on Source/WebKit/WebProcess/WebPage/Cocoa/WebPageCocoa.mm:
```diff
@@ -301,6 +302,21 @@
+
+void WebPage::updateTextIndicatorStyleVisibilityForID(const WTF::UUID uuid, bool visible, CompletionHandler<void()>&& completionHandler)
+{
+    // FIXME: turn on/off the visibility
```
Nit:
```suggestion
    // FIXME: Turn on/off the visibility.
```

pxlcoder commented on Source/WebKit/WebKitSwift/TextIndicatorStyle/WKSTextStyleManager.h:
```diff
@@ -0,0 +1,50 @@
+@protocol WKSTextStyleSourceDelegate <NSObject>
+- (void)targetedPreviewForID:(NSUUID *)uuid completionHandler:(void (^)(UITargetedPreview * _Nullable))completionHandler;
+- (void)updateTextIndicatorStyleVisibilityForID:(NSUUID *)uuid visible:(BOOL)visible completionHandler:(void (^)(void))completionHandler;
+- (UIView *)containingView;
```
This is too generic of a name to implement on a `UIView` subclass. Maybe `containingViewForTextIndicatorStyle`?

rr-codes commented on Source/WebKit/WebProcess/WebPage/Cocoa/UnifiedTextReplacementController.mm:
```diff
@@ -699,6 +704,48 @@ static void replaceContentsInRange(WebCore::Document& document, const WebCore::S
+        return;
+    }
+
+    textIndicatorData = textIndicator->data();
```
the declaration of textIndicatorData can be moved to here, and it doesn’t need to be optional anymore

pxlcoder commented on Source/WebKit/UIProcess/WebPageProxy.h:
```diff
@@ -38,6 +38,7 @@
 
 #if USE(DICTATION_ALTERNATIVES)
 #include <WebCore/PlatformTextAlternatives.h>
+#include <WebCore/SimpleRange.h>
```
This include seems unnecessary.

pxlcoder commented on Source/WebKit/WebKit.xcodeproj/project.pbxproj:
```diff
@@ -20541,20 +20564,27 @@
 			baseConfigurationReference = CD09FD0A26152E2300E4ACF1 /* WebKitSwift.xcconfig */;
 			buildSettings = {
+				CLANG_ENABLE_MODULES = YES;
+				SWIFT_VERSION = 5.0;
```
Someone else should review this part.

megangardner committed:
Add support for text indicator style.
https://bugs.webkit.org/show_bug.cgi?id=271877
rdar://123134420

Reviewed by Aditya Keerthi.

Add support for text indicator style.

* Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
* Source/WebKit/Configurations/WebKitSwift.xcconfig:
* Source/WebKit/DerivedSources-input.xcfilelist:
* Source/WebKit/DerivedSources-output.xcfilelist:
* Source/WebKit/DerivedSources.make:
* Source/WebKit/UIProcess/API/Cocoa/WKWebView.mm:
(-[WKWebView _removeStyleForID:]):
* Source/WebKit/UIProcess/API/Cocoa/WKWebViewInternal.h:
* Source/WebKit/UIProcess/Cocoa/PageClientImplCocoa.h:
* Source/WebKit/UIProcess/Cocoa/PageClientImplCocoa.mm:
(WebKit::PageClientImplCocoa::removeStyleForID):
* Source/WebKit/UIProcess/Cocoa/WebKitSwiftSoftLink.h:
* Source/WebKit/UIProcess/Cocoa/WebKitSwiftSoftLink.mm:
* Source/WebKit/UIProcess/Cocoa/WebPageProxyCocoa.mm:
(WebKit::WebPageProxy::getTextIndicatorForID):
(WebKit::WebPageProxy::updateTextStyleVisibilityForID):
* Source/WebKit/UIProcess/PageClient.h:
* Source/WebKit/UIProcess/WebPageProxy.cpp:
(WebKit::WebPageProxy::removeStyleForID):
* Source/WebKit/UIProcess/WebPageProxy.h:
* Source/WebKit/UIProcess/WebPageProxy.messages.in:
* Source/WebKit/UIProcess/ios/WKContentView.h:
* Source/WebKit/UIProcess/ios/WKContentViewInteraction.h:
* Source/WebKit/UIProcess/ios/WKContentViewInteraction.mm:
(-[WKContentView addStyleForID:]):
(-[WKContentView removeStyleForID:]):
(-[WKContentView targetedPreviewForID:completionHandler:]):
(-[WKContentView updateTextStyleVisibilityForID:visible:completionHandler:]):
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
* Source/WebKit/WebKitSwift/TextIndicatorStyle/WKSTextStyleManager.h: Copied from Source/WebKit/WebKitSwift/WebKitSwift.h.
* Source/WebKit/WebKitSwift/WebKitSwift.h:
* Source/WebKit/WebProcess/WebCoreSupport/WebChromeClient.h:
* Source/WebKit/WebProcess/WebPage/Cocoa/UnifiedTextReplacementController.mm:
(WebKit::UnifiedTextReplacementController::textReplacementSessionDidReceiveReplacements):
(WebKit::UnifiedTextReplacementController::textReplacementSessionDidReceiveTextWithReplacementRange):
* Source/WebKit/WebProcess/WebPage/Cocoa/WebPageCocoa.mm:
(WebKit::WebPage::getTextIndicatorForID):
(WebKit::WebPage::updateTextStyleVisibilityForID):
* Source/WebKit/WebProcess/WebPage/UnifiedTextReplacementController.h:
* Source/WebKit/WebProcess/WebPage/WebPage.cpp:
(WebKit::WebPage::removeStyleForID):
* Source/WebKit/WebProcess/WebPage/WebPage.h:
* Source/WebKit/WebProcess/WebPage/WebPage.messages.in:

Canonical link: https://commits.webkit.org/276943@main
